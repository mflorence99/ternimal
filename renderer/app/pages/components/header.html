<header>

  <ternimal-tabs
    class="tabs">
  </ternimal-tabs>

  <ternimal-system-info
    class="system-info">
  </ternimal-system-info>

  <ternimal-toolbar
    class="toolbar">
  </ternimal-toolbar>

  <div
    class="border">

    <div
      class="sub-tab">
    </div>

  </div>

</header>
