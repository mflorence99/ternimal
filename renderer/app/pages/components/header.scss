:host {
  display: block;
}

header {
  background-color: var(--mat-grey-850);
  display: grid;
  grid-column-gap: 2rem;
  grid-template-columns: 1fr minmax(120px, 0.25fr) auto;
  grid-template-rows: 1fr 5px;
  height: 100%;
}

.border {
  background-color: var(--mat-grey-800);
  display: flex;
  flex-direction: row;
  grid-column: 1 / span 3;
  grid-row: 2;
  height: 100%;

  .sub-tab {
    background-color: var(--primary-color);
    height: 100%;
    width: 14rem;
  }

}
