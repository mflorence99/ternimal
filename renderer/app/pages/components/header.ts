import { SelectionState } from '../../state/selection';
import { Tab } from '../../state/tabs';
import { TabsComponent } from './tabs';

import { ChangeDetectionStrategy } from '@angular/core';
import { Component } from '@angular/core';
import { ViewChild } from '@angular/core';

@Component({
  changeDetection: ChangeDetectionStrategy.OnPush,
  selector: 'ternimal-header',
  templateUrl: 'header.html',
  styleUrls: ['header.scss']
})
export class HeaderComponent {
  @ViewChild(TabsComponent, { static: true }) tabs: TabsComponent;

  constructor(public selection: SelectionState) {}

  isMoreSelected(): boolean {
    return this.tabs.inMore.some(
      (tab) => tab.layoutID === this.selection.layoutID
    );
  }

  isTabsSelected(tab: Tab): boolean {
    return tab.layoutID === this.selection.layoutID;
  }
}
