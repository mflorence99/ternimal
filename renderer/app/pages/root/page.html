  <mat-sidenav-container
    class="container">

    <mat-sidenav 
      [opened]="ternimal.tabPrefsShowing"
      class="sidenav"
      mode="side"
      position="end">

      <ternimal-tab-prefs>
      </ternimal-tab-prefs>
      
    </mat-sidenav>

    <mat-sidenav-content
      class="content">

      <ternimal-header
        class="header">
      </ternimal-header>

      <ternimal-splittable
        [splittable]="layout.layout"
        class="splittable">
      </ternimal-splittable>

    </mat-sidenav-content>

  </mat-sidenav-container>
