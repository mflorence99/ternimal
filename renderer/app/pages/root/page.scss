:host {
  --ternimal-cy-header: 48px;

  display: block;
  height: 100%;
  position: relative;
  width: 100%;
}

.header {
  display: block;
  height: var(--ternimal-cy-header);
  position: absolute;
  top: 0;
  width: 100%;
}
