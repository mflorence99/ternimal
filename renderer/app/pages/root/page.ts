import { LayoutState } from '../../state/layout';
import { TernimalState } from '../../state/ternimal';

import { ChangeDetectionStrategy } from '@angular/core';
import { Component } from '@angular/core';

@Component({
  changeDetection: ChangeDetectionStrategy.Default,
  selector: 'ternimal-root',
  templateUrl: 'page.html',
  styleUrls: ['page.scss']
})

export class RootPageComponent {

  /** ctor */
  constructor(public layout: LayoutState,
              public ternimal: TernimalState) { }

}
