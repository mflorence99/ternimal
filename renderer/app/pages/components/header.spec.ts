import { HeaderComponent } from './header';

import { prepare } from '../page.spec';

import 'jest-extended';

import { ComponentFixture } from '@angular/core/testing';
import { TestBed } from '@angular/core/testing';

describe('RootComponent', () => {
  let component: HeaderComponent;
  let fixture: ComponentFixture<HeaderComponent>;

  beforeEach(() => {
    prepare();
    fixture = TestBed.createComponent(HeaderComponent);
    component = fixture.componentInstance;
  });

  test('isMoreSelected', () => {
    component.selection.selectLayout({ layoutID: 'l' });
    component.tabs.inMore = [{ layoutID: 'l' }, { layoutID: 'm' }];
    expect(component.isMoreSelected()).toBeTrue();
    component.tabs.inMore = [{ layoutID: 'm' }];
    expect(component.isMoreSelected()).toBeFalse();
  });

  test('isTabsSelected', () => {
    component.selection.selectLayout({ layoutID: 'l' });
    expect(component.isTabsSelected({ layoutID: 'l' })).toBeTrue();
    expect(component.isTabsSelected({ layoutID: 'm' })).toBeFalse();
  });

  test('snapshot', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
