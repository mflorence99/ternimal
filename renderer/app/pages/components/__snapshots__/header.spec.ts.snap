// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RootComponent snapshot 1`] = `
<ternimal-header
  __ngContext__={[Function LRootView]}
  selection={[Function SelectionState]}
  tabs={[Function TabsComponent]}
>
  <header
    class="header"
  >
    <ternimal-tabs>
      <nav
        cdkdroplist=""
        cdkdroplistorientation="horizontal"
        class="cdk-drop-list tabs"
      >
        
        
      </nav>
    </ternimal-tabs>
    <ternimal-system-info>
      <article
        class="system-info"
      >
        <ternimal-eq-gauge
          tag="CPU"
        >
          <figure
            class="eq-gauge monospace"
          >
            <figcaption
              class="tag"
            >
              CPU
            </figcaption>
            <ul
              class="gauge"
            >
              
            </ul>
            <figcaption
              class="value"
            >
              0%
            </figcaption>
          </figure>
        </ternimal-eq-gauge>
        <div
          class="spacer"
        />
        <ternimal-eq-gauge
          tag="Mem"
        >
          <figure
            class="eq-gauge monospace"
          >
            <figcaption
              class="tag"
            >
              Mem
            </figcaption>
            <ul
              class="gauge"
            >
              
            </ul>
            <figcaption
              class="value"
            >
              0%
            </figcaption>
          </figure>
        </ternimal-eq-gauge>
      </article>
    </ternimal-system-info>
    <ternimal-toolbar>
      <menu
        class="toolbar"
      >
        <button
          aria-describedby="cdk-describedby-message-4"
          cdk-describedby-host=""
          class="mat-focus-indicator mat-tooltip-trigger mat-icon-button mat-button-base"
          mat-icon-button=""
          mattooltip="Add a new tab"
        >
          <span
            class="mat-button-wrapper"
          >
            <fa-icon
              class="ng-fa-icon"
              size="lg"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-plus fa-w-14 fa-lg"
                data-icon="plus"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                  fill="currentColor"
                />
              </svg>
            </fa-icon>
          </span>
          <span
            class="mat-ripple mat-button-ripple mat-button-ripple-round"
            matripple=""
          />
          <span
            class="mat-button-focus-overlay"
          />
        </button>
        
        <button
          class="mat-focus-indicator mat-tooltip-trigger mat-icon-button mat-button-base"
          mat-icon-button=""
        >
          <span
            class="mat-button-wrapper"
          >
            <fa-icon
              class="ng-fa-icon"
              size="lg"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-cog fa-w-16 fa-lg"
                data-icon="cog"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
                  fill="currentColor"
                />
              </svg>
            </fa-icon>
          </span>
          <span
            class="mat-ripple mat-button-ripple mat-button-ripple-round"
            matripple=""
          />
          <span
            class="mat-button-focus-overlay"
          />
        </button>
        
        <button
          aria-describedby="cdk-describedby-message-5"
          cdk-describedby-host=""
          class="mat-focus-indicator mat-tooltip-trigger mat-icon-button mat-button-base"
          mat-icon-button=""
          mattooltip="More actions..."
        >
          <span
            class="mat-button-wrapper"
          >
            <fa-icon
              class="ng-fa-icon"
              size="lg"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-ellipsis-h fa-w-16 fa-lg"
                data-icon="ellipsis-h"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M328 256c0 39.8-32.2 72-72 72s-72-32.2-72-72 32.2-72 72-72 72 32.2 72 72zm104-72c-39.8 0-72 32.2-72 72s32.2 72 72 72 72-32.2 72-72-32.2-72-72-72zm-352 0c-39.8 0-72 32.2-72 72s32.2 72 72 72 72-32.2 72-72-32.2-72-72-72z"
                  fill="currentColor"
                />
              </svg>
            </fa-icon>
          </span>
          <span
            class="mat-ripple mat-button-ripple mat-button-ripple-round"
            matripple=""
          />
          <span
            class="mat-button-focus-overlay"
          />
        </button>
        
      </menu>
      <context-menu />
    </ternimal-toolbar>
    <ul
      class="border"
    >
      
      
    </ul>
  </header>
</ternimal-header>
`;
